### DISCUSSION TITLE
@ngrx/signals DeepSignal not created when patching state

### DISCUSSION THREAD

**Discussion by k3nsei** [ngrx/platform#4494] [General]
### Which @ngrx/* package(s) are relevant/related to the feature request?

signals

### Information

`DeepSignals` are only created when defining initial state using `withState`. When I'm using `patchState` inside `onInit` hook `DeepSignal` isn't created.

I want to be able to set empty object as placeholder inside state, and then in `onInit` hook replace this placeholder with result of factory function injected by dependency injection (factory fn takes reference to store as argument). Result from factory would be patched into state. I'm expecting it to be `DeepSignal`, but it isn't. It's worth to note that it is typed by signal store as `DeepSignal`.

Real world usage could be find here: https://github.com/k3nsei/ngx-signal-store-query/blob/main/libs/ngx-signal-store-query/src/with-query.ts

So here https://github.com/k3nsei/ngx-signal-store-query/blob/1cb2edb83ddd46ed6e3017e3d75d7714d31e572e/apps/demo/src/app/app.component.ts#L49 instead of current implementation, it would be nice to be able to use something like:

```TypeScript
protected readonly data = this.githubQuery.data;
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by rainerhahnekamp** [comment]
@ k3nsei, could you create a minimal and reproducible example on StackBlitz? If some investigation is required, that would help us a lot.

Here's the template: https://stackblitz.com/edit/ngrx-seed

**Comment by k3nsei** [comment]
Provided seed was for NGRX so I created my own here as issue is related to Signal Store.

https://stackblitz.com/edit/stackblitz-ng-signal-sore-9aq8nt-fvbemt?file=src%2Fmain.ts

**Comment by rainerhahnekamp** [comment]
@k3nsei Sorry, that's the official one https://stackblitz.com/github/ngrx/signal-store-starter?file=src%2Fmain.ts but if you have created your one own, that's fine as well. 

Btw. both seeders are shown when you create a new issue

**Comment by rainerhahnekamp** [comment]
@k3nsei I am going to move this to discussion, okay? If it turns out we have here a bug or feature request, we can create a new one.

**Comment by rainerhahnekamp** [comment]
Please correct me if I'm wrong. You need to dynamically initialize the state, and the initial state comes from a Service? That's why you do it in `onInit`?

Because of that dynamic behavior, the state cannot provide the right slices?

There was a similar discussion in https://github.com/ngrx/platform/discussions/4341 but I'm not sure if that is the same as yours.

**Comment by k3nsei** [comment]
Ok, I solved my problem with a different approach. I used a mix of `withMethods` and a wrapped method into `Proxy`. It is quite advanced black magic and super hacky, but it works 😝 

Here is example of my working solution https://github.com/k3nsei/ngx-signal-store-query/blob/main/libs/ngx-signal-store-query/src/with-query.ts
